### ISSUE TITLE
R6 method chaining

### ISSUE THREAD

**Issue by b-rhodes** [r-lib/lintr#2813]
Is there a way to get the linter to recognize R6 method chaining and not complain?

[https://adv-r.hadley.nz/r6.html#method-chaining](https://adv-r.hadley.nz/r6.html#method-chaining)

When I try to lay out code like this:
```R
x$
  add(10)$
  add(10)$
  sum
```

It complains about incorrect numbers of indents.

**Comment by MichaelChirico** [comment]
Could you provide a reproducible example? I'm not seeing any lints from copy-pasting yours:

```r
l = modify_defaults(all_linters(), implicit_integer_linter = NULL)
lint("
x$
  add(10)$
  add(10)$
  sum
", linters = l)
# ℹ No lints found.
```

**Comment by b-rhodes** [comment]
Sorry! I didn't realise having or not method parameters would make a difference.
```R
r$> l = modify_defaults(all_linters(), implicit_integer_linter = NULL)
    lint("
    x$
      add()$
      add()$
      sum
    ", linters = l)

# Output:
<text>:2:2: warning: [extraction_operator_linter] Use `[[` instead of `$` to extract an element.
x$
 ^
<text>:3:2: style: [indentation_linter] Indentation should be 0 spaces but is 2 spaces.
  add()$
~^
<text>:3:8: warning: [extraction_operator_linter] Use `[[` instead of `$` to extract an element.
  add()$
       ^
<text>:4:8: warning: [extraction_operator_linter] Use `[[` instead of `$` to extract an element.
  add()$
       ^
```

**Comment by MichaelChirico** [comment]
can you try updating to current dev version? extraction_operator_linter() is deprecated so I suspect you're on an old version 

https://lintr.r-lib.org/reference/lintr-deprecated.html

**Comment by b-rhodes** [comment]
Updating to the newest version on CRAN (3.2.0) changes the result to this:
```R
r$> l = modify_defaults(all_linters(), implicit_integer_linter = NULL)
    lint("
    x$
      add()$
      add()$
      sum
    ", linters = l)
<text>:3:2: style: [indentation_linter] Indentation should be 0 spaces but is 2 spaces.
  add()$
~^
```

**Comment by MichaelChirico** [comment]
Thanks, I confirm the issue. Not sure why my earlier example didn't spot it.
